src/main/java/org/primefaces/component/terminal/TerminalRenderer.java

```diff
@@ -102,17 +110,40 @@ protected void encodeScript(FacesContext context, Terminal terminal) throws IOEx
     }
     
     protected void handleCommand(FacesContext context, Terminal terminal) throws IOException {
-        ResponseWriter writer = context.getResponseWriter();
-        String clientId = terminal.getClientId(context);
-        String value = context.getExternalContext().getRequestParameterMap().get(clientId + "_input");
-        String tokens[] = value.split(" ");
+        String tokens[] = getValueTokens(context, terminal);
         String command = tokens[0];
         String[] args = Arrays.copyOfRange(tokens, 1, tokens.length);
 
         MethodExpression commandHandler = terminal.getCommandHandler();
         String result = (String) commandHandler.invoke(context.getELContext(), new Object[]{command, args});
 
+        ResponseWriter writer = context.getResponseWriter();
         writer.write(result);
```

Comment:
potential XSS

Reply:
Sorry about blaming this pull request. This XSS vulnerability has already existed before. I created an issue: https://github.com/primefaces/primefaces/issues/3623